samza-hdfs/src/main/java/org/apache/samza/system/hdfs/HdfsSystemAdmin.java

```diff
@@ -116,6 +116,13 @@ public HdfsSystemAdmin(String systemName, Config config) {
   }
 
   static Map<Partition, List<String>> obtainPartitionDescriptorMap(String stagingDirectory, String streamName) {
+    if (StringUtils.isBlank(stagingDirectory)) {
+      LOG.warn("Empty staging directory");
+      return null;
```

Comment:
+1. Please also change the logging to info since it's expected when runner only needs to query the meta data.